restli-server/src/main/java/com/linkedin/restli/server/ResourceDefinition.java

```diff
@@ -52,6 +52,12 @@
    */
   String getD2ServiceName();
 
+  /**
+   * Gets the base uri template.
+   * @return the base uri template.
+   */
+  String getBaseUriTemplate();
```

Comment:
Should we add a default implementation here to prevent build incompatibilities?

Reply:
There is only one implementation available (ResourceModel). Do you think it may be extended externally? Or is this just being extra safe?

Reply:
I searched and didn't find any other internal usages. It's likely safe, however our policy is that we follow semver, so I'd suggest you just provide a default implementation that throws a `NotImplementedException` (or whatever it's called)

Reply:
Done